### PR TITLE
merge nov 20th - update sql linting to ga

### PR DESCRIPTION
**Pull request by mirnawong1**
SQL linting for Ci jobs will soon be GA. This pr removes any beta callout and also enhances the sql linting section so it provides users add'l links and steps on how to configure. 

To merge on Nov 20th

[docs notion](https://www.notion.so/dbtlabs/Orc-Linting-in-CI-GA-13cbb38ebda78043bd8bf66cd0805b9c?pvs=4)

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-linting-ci-update-dbt-labs.vercel.app/docs/cloud/dbt-cloud-ide/lint-format
- https://docs-getdbt-com-git-linting-ci-update-dbt-labs.vercel.app/docs/dbt-versions/release-notes
- https://docs-getdbt-com-git-linting-ci-update-dbt-labs.vercel.app/docs/deploy/ci-jobs
- https://docs-getdbt-com-git-linting-ci-update-dbt-labs.vercel.app/docs/deploy/continuous-integration

### CODE REVIEW

File: website/docs/docs/dbt-versions/release-notes.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Release notes are grouped by month for both multi-tenant and virtual private clo
 \* The official release date for this new format of release notes is May 15th, 2024. Historical release notes for prior dates may not reflect all available features released earlier this year or their tenancy availability.
 
 ## November 2024
+- **New**: SQL linting in CI jobs is now generally available in dbt Cloud. You can enable SQL linting in your CI jobs, using [SQLFluff](https://sqlfluff.com/), to automatically lint all SQL files in your project as a run step before your CI job builds. Refer to [SQL linting](/docs/deploy/continuous-integration#sql-linting) for more information.
```
Comment (mirnawong1) [inline website/docs/docs/dbt-versions/release-notes.md]: @reubenmc any specific requirements to add here? versionless? plan? based on the release change plan, it'll be available to all tenants/mc.

Comment (mikaylacrawford) [reply website/docs/docs/dbt-versions/release-notes.md]: There's extra restrictions that it's only available for versionless CI jobs & for enterprise or team plans

File: website/docs/docs/deploy/continuous-integration.md [hunk 2]
```diff
@@ -56,10 +56,18 @@ When you push a new commit to a PR, dbt Cloud enqueues a new CI run for the late
...
+If the linter runs into errors, you can specify whether dbt should stop running the job on error or continue running it on error. When failing jobs, it helps reduce compute costs by avoiding builds for pull requests that don't meet your SQL code quality CI check.
+
+#### To configure SQLFluff linting:
+
+- Use [SQLFluff Configuration Files](https://docs.sqlfluff.com/en/stable/configuration/setting_configuration.html#configuration-files) to override the default linting behavior in dbt.
+- Create a `.sqlfluff` configuration file in your project, add your linting rules to it, and dbt Cloud will use them when linting.
+    - When configuring, you can use `dbt` as the templater (for example, `templater = dbt`)
+    - If you’re using the dbt Cloud IDE, dbt Cloud CLI, or any other editor, refer to [Customize linting](/docs/cloud/dbt-cloud-ide/lint-format#customize-linting) for guidance on how to add dbt-specific (or dbtonic) linting rules we use for own project.
```
Comment (mirnawong1) [inline website/docs/docs/deploy/continuous-integration.md]:
```suggestion
    - If you’re using the dbt Cloud IDE, dbt Cloud CLI, or any other editor, refer to [Customize linting](/docs/cloud/dbt-cloud-ide/lint-format#customize-linting) for guidance on how to add the dbt-specific (or dbtonic) linting rules we use for own project.
```

**Review by nataliefiann — APPROVED** [review]
Hiya @mirnawong1 

Thanks for creating this PR! it's great. I've approved this for you with no suggestions or recommendations.

Kind Regards
Natalie

File: website/docs/docs/deploy/continuous-integration.md [hunk 3]
```diff
@@ -56,10 +56,18 @@ When you push a new commit to a PR, dbt Cloud enqueues a new CI run for the late
...
 
-You can use [SQLFluff Configuration Files](https://docs.sqlfluff.com/en/stable/configuration/setting_configuration.html#configuration-files) to override the default linting behavior in dbt. Create an `.sqlfluff` configuration file in your project, add your linting rules to it, and dbt Cloud will use them when linting. For complete details, refer to [Custom Usage](https://docs.sqlfluff.com/en/stable/gettingstarted.html#custom-usage) in the SQLFluff documentation. 
+If the linter runs into errors, you can specify whether dbt should stop running the job on error or continue running it on error. When failing jobs, it helps reduce compute costs by avoiding builds for pull requests that don't meet your SQL code quality CI check.
+
+#### To configure SQLFluff linting:
+
+- Use [SQLFluff Configuration Files](https://docs.sqlfluff.com/en/stable/configuration/setting_configuration.html#configuration-files) to override the default linting behavior in dbt.
```
Comment (reubenmc) [inline website/docs/docs/deploy/continuous-integration.md:70]: It is clear enough that having a .sqlfluff file is optional? And people can simply use it without one if they want?

**Review by reubenmc — APPROVED** [review]
Small comment, but LGTM!

File: website/docs/docs/deploy/continuous-integration.md (hunk 3 above)
Comment (mirnawong1) [inline website/docs/docs/deploy/continuous-integration.md:70]:
```suggestion
- (Optional) Use [SQLFluff Configuration Files](https://docs.sqlfluff.com/en/stable/configuration/setting_configuration.html#configuration-files) to override the default linting behavior in dbt.
```

**Pull request merged**